File: convert_geo.py
import re

file_path = 'c:/Users/jlima/Desktop/REPORTE DE ASESORES/geo_hierarchy.js'

with open(file_path, 'r', encoding='utf-8') as f:
    lines = f.readlines()

estado = ""
municipio = ""
parroquia = ""
in_sectors = False
values = []

for line in lines:
    line = line.strip()
    if not line or line.startswith('export') or line.startswith('};'):
        continue
    
    # State level: "Aragua": {
    m = re.match(r'^"([^"]+)":\s*\{$', line)
    if m:
        if not estado:
            estado = m.group(1)
        elif not municipio:
            municipio = m.group(1)
        continue
    
    # Parroquia/Sector list level: "El Consejo": [
    m = re.match(r'^"([^"]+)":\s*\[$', line)
    if m:
        parroquia = m.group(1)
        in_sectors = True
        continue
    
    # Sector: "Centro",
    m = re.match(r'^"([^"]+)"(,?)$', line)
    if m and in_sectors:
        sector = m.group(1)
        e = estado.replace("'", "''")
        mu = municipio.replace("'", "''")
        p = parroquia.replace("'", "''")
        s = sector.replace("'", "''")
        values.append(f"('{e}', '{mu}', '{p}', '{s}')")
        continue
    
    # End of sector list: ],
    if line.startswith(']'):
        in_sectors = False
        continue
        
    # End of municipio or estado: },
    if line.startswith('}'):
        if municipio and not in_sectors:
            # municipio = "" -- Wait, we need to know if we are closing a municipio or a state
            # Simple hack: check if next line is a quote or a closing brace
            pass
        # A more robust way is to check the indentation in the original file, but we stripped it
        continue

# Alternative approach: use the indent
indent_values = []
estado = ""
municipio = ""
parroquia = ""

with open(file_path, 'r', encoding='utf-8') as f:
    for line in f:
        stripped = line.strip()
        if not stripped: continue
        indent = len(line) - len(line.lstrip())
        
        m_obj = re.match(r'^"([^"]+)":\s*\{', stripped)
        m_arr = re.match(r'^"([^"]+)":\s*\[', stripped)
        m_str = re.match(r'^"([^"]+)"', stripped)
        
        if m_obj:
            name = m_obj.group(1)
            if indent == 2: estado = name
            if indent == 4: municipio = name
        elif m_arr:
            parroquia = m_arr.group(1)
        elif m_str and indent == 8:
            sector = m_str.group(1)
            e = estado.replace("'", "''")
            mu = municipio.replace("'", "''")
            p = parroquia.replace("'", "''")
            s = sector.replace("'", "''")
            indent_values.append(f"('{e}', '{mu}', '{p}', '{s}')")

sql = f"INSERT INTO geodata_config (estado, municipio, parroquia, sector) VALUES \n" + ",\n".join(indent_values) + ";"

with open('c:/Users/jlima/Desktop/REPORTE DE ASESORES/geo_inserts.sql', 'w', encoding='utf-8') as f:
    f.write(sql)

print(f"Generated {len(indent_values)} inserts.")

File: convert_geo.js
const fs = require('fs');

const content = fs.readFileSync('c:/Users/jlima/Desktop/REPORTE DE ASESORES/geo_hierarchy.js', 'utf8');

// Use regex to find all matches of "Level": { or "Level": [
// and tracking the current context.

const lines = content.split('\n');
let estado = '';
let municipio = '';
let parroquia = '';
let values = [];

for (let line of lines) {
    let trimmed = line.trim();
    if (!trimmed) continue;
    
    // Indentation check
    let indent = line.search(/\S/);
    
    // State: "Aragua": { (indent 2)
    let m_state = trimmed.match(/^"([^"]+)":\s*\{/);
    if (m_state && indent === 2) {
        estado = m_state[1];
        continue;
    }
    
    // Municipio: "José Rafael Revenga": { (indent 4)
    let m_mun = trimmed.match(/^"([^"]+)":\s*\{/);
    if (m_mun && indent === 4) {
        municipio = m_mun[1];
        continue;
    }
    
    // Parroquia: "El Consejo": [ (indent 6)
    let m_par = trimmed.match(/^"([^"]+)":\s*\[/);
    if (m_par && indent === 6) {
        parroquia = m_par[1];
        continue;
    }
    
    // Sector: "Centro", (indent 8)
    let m_sec = trimmed.match(/^"([^"]+)"/);
    if (m_sec && indent === 8) {
        let sector = m_sec[1];
        let e = estado.replace(/'/g, "''");
        let mu = municipio.replace(/'/g, "''");
        let p = parroquia.replace(/'/g, "''");
        let s = sector.replace(/'/g, "''");
        values.push(`('${e}', '${mu}', '${p}', '${s}')`);
    }
}

const sql = `INSERT INTO geodata_config (estado, municipio, parroquia, sector) VALUES \n${values.join(',\n')};`;
fs.writeFileSync('c:/Users/jlima/Desktop/REPORTE DE ASESORES/geo_inserts.sql', sql);
console.log(`Successfully generated ${values.length} inserts.`);
